### PR TITLE
fix/define autochecked checkboxes on issue list in firefox

### LINKED ISSUE #7311: Closing issue(s) by using batch checkbox leaves wrong item ticked.
- Gitea version (or commit ref):  1.9.0+dev-382-gc37ec66ee 
- Operating system: Linux (Devuan)
- Browser: FireFox
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

Closing issue(s) by using batch checkbox leaves wrong item ticked. 

## Reproduce

1. View a list of OPEN issues with more than 1 issue.
2. Use the batch operations checkboxes to select the FIRST item
3. Use the batch operations close button to close the issue

## Expected
The issue is closed, and is removed from the list. No other issues are ticked.

## Actual
The issue is close and removed as expected, however the position it was at remains ticked. As the issues have all moved up to fill the space, now an unrelated issue is ticked.

## Problem
A second click of the close button could close the wrong issue. Or if we wish to do another batch operation we need to untick all the wrong ticks.

## Fix
Untick all checkboxes after batch operation, but only if the operation was CLOSE.

I can see this is useful to keep them ticked if we are modifying milestone or label as we may want to preform another operation on the same issues, so I don't think we want to remove this functionality completely.

### PR DESCRIPTION
**Pull request by Cherrg**
- close/open: deselect all
- nonclose: show buttons for close/label/milestone after reload

fix #7311

Signed-off-by: Michael Gnehr <michael@gnehr.de>

#### Note
As Webkit browsers do not have the bug because they do not cache the checkbox states, this PR is targeting Firefox only

#### Note2
If the cache behavior is also wanted on other browsers, it should be possible to implement via localstorage. Leave a comment if i should add this, or merge/close this issue if not.

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -2106,13 +2106,27 @@ $(document).ready(function () {
...
         }).get().join();
-        var url = this.dataset.url
-        updateIssuesMeta(url, action, issueIDs, elementId).then(reload);
+        let url = this.dataset.url;
+        updateIssuesMeta(url, action, issueIDs, elementId).then(function() {
+            if (action === "close" || action === "open" ){
+                //uncheck all checkboxes
+                $('.issue-checkbox input[type="checkbox"]').each(function(_,e){ e.checked = false; });
+            }
+            setTimeout(reload, 5);
```
Comment (silverwind) [inline public/js/index.js]: Why these arbitrary timeouts?

Comment (Cherrg) [reply public/js/index.js]: added them while debugging, forgot to remove them

-> removed now

**Comment by silverwind** [comment]
Do you think it would be hard to eliminate the page reload altogether? That'd be my preferred solution but if it's too much work, we can land this now, but please include a comment that this is a workaround targeting Firefox only.

**Comment by Cherrg** [comment]
it should be possible to remove the reload for the ui actions. The task will be updating the ui with js instead of the go template engine.

#### Tasklist:
* remove items on `open`/`delete` -> easy
* remove labels -> easy
* add labels -> color and name is there, but description is missing on current ui -> need to be added inside template first
* remove milestones -> easy
* update/add milestones should be easy too
* assignee -> all nessesary information should be accessible via js -> easy too

- may it is nessesary to update/sync current label and milestone list from server from time to time

In my optinion this should be done step by step and in extra issue.
Furthermore i found a bug on issue list view: reset of assignees is not possible and result in server 500 error.

**Comment by Cherrg** [comment]
updated merge description -> add comment this PR only target FF (as bug is there present only)

**Comment by Cherrg** [comment]
i added issues for both: #7328 #7329

**Comment by silverwind** [comment]
> In my optinion this should be done in an extra issue.

Agree, it does sound like a lot of work actually.

**Review by silverwind — APPROVED** [review]
LGTM, but would still appreciate a code comment mentioning that this is a workaround for Firefox.

**Review by mrsdizzie — APPROVED** [review]

**Comment by lafriks** [comment]
Conflict

**Comment by Cherrg** [comment]
@silverwind 
> LGTM, but would still appreciate a code comment mentioning that this is a workaround for Firefox.

added

**Comment by Cherrg** [comment]
resolved confilct

**Pull request merged**